package/gluon-mesh-babel/src/respondd.c

```diff
@@ -71,13 +75,250 @@
 #define VPN_INTERFACE "mesh-vpn"
 #define l3rdctl "/var/run/l3roamd.sock"
 
-#define IFNAMELEN 32
 #define PROTOLEN 32
 
 #define UBUS_TIMEOUT 30
 #define UBUS_SOCKET "/var/run/ubus.sock"
 
 static struct babelhelper_ctx bhelper_ctx = {};
+static struct json_object *babeld_version = NULL;
+
+static char *model = NULL;
+static struct json_object *neighbours = NULL;
+static pthread_mutex_t neighbours_lock;
+static pthread_t babelmonitor;
+
+struct thread_info {
+	pthread_t thread_id;
+	int thread_num;
+	char *argv_string;
+};
+
+struct kernel_route {
+	struct in6_addr prefix;
+	struct in6_addr src_prefix;
+	struct in6_addr gw;
+	int plen;
+	int src_plen; /* no source prefix <=> src_plen == 0 */
+	int metric;
+	int proto;
+	unsigned int ifindex;
+	unsigned int table;
+};
+
+struct nlrtreq {
+	struct nlmsghdr nl;
+	struct rtmsg rt;
+	char buf[1024];
+};
+
+#define ROUTE_PROTO 158
+
+#define STRBUFELEMENTLEN 65
+#define KERNEL_INFINITY 9999
+static char strbuf[STRBUFELEMENTLEN];
+
+static const char *print_ip(const struct in6_addr *addr) {
+	return inet_ntop(AF_INET6, &(addr->s6_addr), strbuf, STRBUFELEMENTLEN);
+}
+
+static int rtnl_addattr(struct nlmsghdr *n, size_t maxlen, int type, void *data, int datalen) {
```

Comment:
`nla_put()` should work, too. I think.

Reply:
There are a lot of places where things from `libnl-tiny` could have been used

Reply:
I am not sure what to do with this review comment. I used the pattern here as in other projects that I have found to work. Of course it could be reworked. Is the benefit significant?

Reply:
If it works, probably not. Something for future developments.

Reply:
ok. thank you for the clarification.